docs/control-structures.rst

```diff
@@ -153,13 +153,12 @@ Assigning *to* a state variable always creates an independent copy. On the other
 Scoping and Declarations
 ========================
 
-In Solidity, a variable which is declared is automatically assigned its default value. It will be assigned on contract
-initialization if it is a contract-level variable or at the beginning of a function call if it is a local variable.
-This is because the EVM must run deterministically so it would be inappropriate to initialize any variables to random garbage values.
+A variable which is declared will have an initial default value whose byte-representation is all zeros.
+This is because the EVM must run without undefined behaviour so it would be inappropriate to initialize any variables to random garbage values.
```

Comment:
The EVM itself will still run without undefined behaviour. I think it is less confusing if this sentence is just removed.